<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="1" />

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />

  <arg name="world_frame" default="$(arg UAV_NAME)/world_origin" />

    <!-- Namespace - all topics, services and parameters will be remapped using this namespace as a prefix (eg.: "/waypoint_flier_simple/odom_uav_in" to "/uav1/waypoint_flier_simple/odom_uav_in") -->
  <group ns="$(arg UAV_NAME)">

    <node pkg="motion_planning_python" type="planner" name="planner" output="screen">

        <!-- ROS parameters config file -->
      <rosparam file="$(find motion_planning_python)/config/motion_planner.yaml" />

        <!-- Voxblox Planner -->
      <!--remap from="~esdf_map_in" to="voxblox_node/esdf_map_out" /-->
      <!--remap from="~tsdf_map_in" to="voxblox_node/tsdf_map_out" /-->
      <remap from="~tsdf_map_in" to="tsdf_map" />
      <param name="use_tf_transforms" value="false" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="update_mesh_every_n_sec" value="0.0" />   
      <param name="publish_traversable" value="false" />
      <param name="publish_pointclouds" value="false" />

        <!-- Publishers -->
      <remap from="~visualization_marker_out" to="visualization_marker_out" />

        <!-- Subscribers -->
      <!--remap from="~constraints_in" to="control_manager/current_constraints" /-->
      <remap from="~uav_state_in" to="estimation_manager/uav_state" />
      <!--remap from="~tracker_cmd_in" to="control_manager/tracker_cmd" /-->
      <remap from="~control_manager_diag_in" to="control_manager/diagnostics" />

        <!-- Services clients-->
      <remap from="~trajectory_generation_out" to="trajectory_generation/get_path" />
      <remap from="~trajectory_reference_out" to="control_manager/trajectory_reference" />
      <!--remap from="~path_out" to="trajectory_generation/path" /-->

        <!-- Services servers-->
      <remap from="~start_in" to="~start" />
      <!--remap from="~planner_type_in" to="~set_planner" /-->
      <!--remap from="~path_in" to="~path" /-->
      <!--remap from="~get_path_in" to="~get_path" /-->

    </node>

  </group>

</launch>
